Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm

```diff
@@ -59,11 +60,226 @@
+                    protectedSelf->_parent = pluginAxObject->wrapper();
+            }
+        });
+    }
+    return protectedSelf->_parent.get().get();
+}
+
+- (void)setParent:(NSObject *)parent
+{
+    _parent = parent;
+}
+
+ALLOW_DEPRECATED_IMPLEMENTATIONS_BEGIN
+- (id)accessibilityAttributeValue:(NSString *)attribute
+ALLOW_DEPRECATED_IMPLEMENTATIONS_END
+{
+    if ([attribute isEqualToString:NSAccessibilityParentAttribute])
+        return [self parent];
+    if ([attribute isEqualToString:NSAccessibilityChildrenAttribute])
+        return [self accessibilityChildren];
+    if ([attribute isEqualToString:NSAccessibilityVisibleChildrenAttribute])
+        return [self accessibilityVisibleChildren];
+    if ([attribute isEqualToString:NSAccessibilityTopLevelUIElementAttribute])
+        return [[self parent] accessibilityAttributeValue:NSAccessibilityTopLevelUIElementAttribute];
+    if ([attribute isEqualToString:NSAccessibilityWindowAttribute])
+        return [[self parent] accessibilityAttributeValue:NSAccessibilityWindowAttribute];
+    if ([attribute isEqualToString:NSAccessibilityEnabledAttribute])
+        return [[self parent] accessibilityAttributeValue:NSAccessibilityEnabledAttribute];
+    if ([attribute isEqualToString:NSAccessibilityRoleAttribute])
+        return NSAccessibilityGroupRole;
+    if ([attribute isEqualToString:NSAccessibilityPrimaryScreenHeightAttribute])
+        return [[self parent] accessibilityAttributeValue:NSAccessibilityPrimaryScreenHeightAttribute];
+    if ([attribute isEqualToString:NSAccessibilitySubroleAttribute])
+        return @"AXPDFPluginSubrole";
+    if ([attribute isEqualToString:NSAccessibilitySizeAttribute])
+    {
+        RefPtr plugin = _pdfPlugin.get();
+        if (plugin)
+            return [NSValue valueWithSize:plugin->boundsOnScreen().size()];
+    }
+    if ([attribute isEqualToString:NSAccessibilityPositionAttribute])
+    {
+        RefPtr plugin = _pdfPlugin.get();
+        if (plugin)
+            return [NSValue valueWithPoint:plugin->boundsOnScreen().location()];
+    }
+    return nil;
+}
+
+ALLOW_DEPRECATED_IMPLEMENTATIONS_BEGIN
+- (NSArray *)accessibilityAttributeNames
+ALLOW_DEPRECATED_IMPLEMENTATIONS_END
+{
+    static NeverDestroyed<RetainPtr<NSArray>> attributeNames = @[
+        NSAccessibilityParentAttribute,
+        NSAccessibilityWindowAttribute,
+        NSAccessibilityTopLevelUIElementAttribute,
+        NSAccessibilityRoleDescriptionAttribute,
+        NSAccessibilitySizeAttribute,
+        NSAccessibilityEnabledAttribute,
+        NSAccessibilityPositionAttribute,
+        NSAccessibilityFocusedAttribute,
+        NSAccessibilityChildrenAttribute,
+        NSAccessibilityPrimaryScreenHeightAttribute,
+        NSAccessibilitySubroleAttribute
+    ];
+    return attributeNames.get().get();
+}
+
+- (BOOL)accessibilityShouldUseUniqueId
+{
+    return YES;
+}
+
+- (NSUInteger)accessibilityArrayAttributeCount:(NSString *)attribute
+{
+    if (!_pdfDocument) {
```

Comment:
This is a bit weird. Why didn't `setPDFDocument:` get called by the plugin?

Reply:
Should be called. But sometimes the pdfdocument is not ready yet when I call it in ‘pluginbase::unit’